### PR TITLE
add support for vagrant-libvirt on GHA

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: .github/workflows/beaker.yml [hunk 1]
```diff
@@ -151,6 +151,11 @@ jobs:
       - name: install additional packages
         if: ${{ inputs.additional_packages != '' }}
         run: sudo apt-get install -y ${{ inputs.additional_packages }}
+      - name: Setup libvirt for Vagrant
+        if: ${{ inputs.beaker_hypervisor == 'vagrant' }}
```
Comment (ekohl) [inline .github/workflows/beaker.yml]: As mentioned in https://github.com/voxpupuli/puppet-mosquitto/pull/50/files#r1461846550, I think this should be `vagrant_libvirt`. `vagrant` is AFAIK Vagrant with VirtualBox.

**Review by bastelfreak — APPROVED** [review]

**Comment by traylenator** [comment]
Trying here - https://github.com/voxpupuli/puppet-swap_file/pull/3 definitely some success 
* almalinux images missing despite https://app.vagrantup.com/almalinux/boxes/9
* Ubunut fails with Call to virDomainCreateWithFlags failed: Invalid machine name

**Comment by ekohl** [comment]
> almalinux images missing despite https://app.vagrantup.com/almalinux/boxes/9

You can see it tries to pull in `generic/almalinux9`. We need to teach beaker-hostgenerator to use the AlmaLinux boxes in https://github.com/voxpupuli/beaker-hostgenerator/blob/master/lib/beaker-hostgenerator/hypervisor/vagrant.rb

> Ubunut fails with Call to virDomainCreateWithFlags failed: Invalid machine name

Somehow this uses a very complex name: `beaker_ubuntu2004-64hostnameubuntu2004-64-puppet7.example.com_62dbfb_ubuntu2004-64-puppet7.example.com`

But for Rocky it does the same: `beaker_rocky8-64hostnamerocky8-64-puppet8.example.com_c59967_rocky8-64-puppet8.example.com`

I don't see other characters that could be disallowed. My only suspicion is the > 100 characters name that could be above a certain limit.

On a related note: it's annoying Vagrant doesn't detect a non-interactive prompt to disable the import progress.

**Comment by ekohl** [comment]
> I don't see other characters that could be disallowed. My only suspicion is the > 100 characters name that could be above a certain limit.

When I try it locally I see it creates `.vagrant/beaker_vagrant_files/beaker_almalinux8-64{hostname=almalinux8-64-puppet8.example.com}_38f6f1/Vagrantfile`:
```ruby
Vagrant.configure("2") do |c|
  c.ssh.insert_key = false
  c.vm.define 'almalinux8-64-puppet8.example.com' do |v|
    v.vm.hostname = 'almalinux8-64-puppet8.example.com'
    v.vm.box = 'almalinux/8'
    v.vm.box_check_update = 'true'
    v.vm.synced_folder '.', '/vagrant', disabled: true
    v.vm.network :private_network, ip: "10.254.215.174", :netmask => "255.255.0.0"
    v.vm.provider :libvirt do |node|
      node.cpus = 1
      node.memory = 1024
      node.qemu_use_session = false

    end
  end
end
```

And that boots up just fine on my Fedora 38.

I wonder if there's a difference in versions or something else. Perhaps the filename is considered in the name and it doesn't like that. You do see it in `beaker_almalinux8-64hostnamealmalinux8-64-puppet8.example.com_1e5c51_almalinux8-64-puppet8.example.com` which is the concatenation of the directory name and the hostname of the actual box (vagrant-libvirt behavior).

**Comment by traylenator** [comment]
https://github.com/vagrant-libvirt/vagrant-libvirt/issues/545 looks like the same thing.

**Comment by ekohl** [comment]
Good news: I can reproduce it in https://github.com/voxpupuli/puppet-selinux/pull/383 so I can test out https://github.com/voxpupuli/beaker-vagrant/pull/80.

**Comment by ekohl** [comment]
confirmed that https://github.com/voxpupuli/beaker-vagrant/pull/80 makes it pass for me: https://github.com/voxpupuli/puppet-selinux/pull/383#issuecomment-1913167116

**Pull request merged**